### PR TITLE
Document PR blocking job requirements

### PR DESCRIPTION
**Pull request by BenTheElder**
#### What type of PR is this:

/kind documentatino

#### What this PR does / why we need it:

The current status quo is that SIG Testing owners of Prow config have informally enforced this every since the introduction of the "release blocking jobs" document, but we've not had a good place to document it. We've been over this many times in slack, github issues, etc., but never captured it properly.

The obvious move is to just append this detail to the release blocking jobs document.

PR blocking jobs already have all of the requirements of release-blocking and the additional requirement that there be high-value in blocking PRs instead of only blocking releases (due to frequent breakage and the high cost of keeping the release blocking signal green without PR coverage, if such a cost exists).

#### Special notes for your reviewer:

Builds on https://github.com/kubernetes/sig-release/pull/2229

/sig testing
/sig release
/sig k8s-infra

### CODE REVIEW

**Comment by BenTheElder** [comment]
/hold
for #2229 and discussion

**Comment by BenTheElder** [comment]
aside: this seems like it should live in k/community (see e.g. https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/vendor.md) or k8s.dev but not a high-prio bikeshed :-)

File: release-blocking-jobs.md [hunk 1]
```diff
@@ -154,3 +155,23 @@ The general process to promote or demote jobs follows these steps:
 
 1. Open a kubernetes/test-infra PR that promotes or demotes the job to/from any
    of the sig-release boards.
+
+## Pull Request Blocking Jobs
```
Comment (aojea) [inline release-blocking-jobs.md:159]: I remember a doc from Aaron with the criteria, so we only have only a subset of selected job, that run in less than one hour, high stability required, ...

Comment (aojea) [reply release-blocking-jobs.md:159]: found it https://github.com/kubernetes/test-infra/issues/6380#issuecomment-887158532

Pasting for reference

```
Let's lift loosely from my https://github.com/kubernetes/sig-release/issues/24#issuecomment-344381137 and https://github.com/kubernetes/sig-release/pull/346 that turned into what [release-blocking criteria are today](https://github.com/kubernetes/sig-release/blob/master/release-blocking-jobs.md#release-blocking-criteria-and-dashboard)

Looking at the merge-blocking jobs we have today, I would hope they meet these criteria:

The job has a 90th percentile duration of 75min (ideally <60min)
The job is better than a coin flip over all PR's (<50% daily failure rate across all PRs)
The job is better than four coin flips against your PR (95% consistent aka 5% flake rate)
The job is supported by contributors who can bring it back to green within a business day if it starts failing continuously or very near so
If the job is e2e, it must use OSS to provision the cluster (no hosted solutions)
```

Comment (aojea) [reply release-blocking-jobs.md:159]: can you incorporate that @BenTheElder ^^^ ?

Comment (BenTheElder) [reply release-blocking-jobs.md:159]: Ooh, thanks for digging that up!

I agree we should include more rigorous criteria and this seems like a good data-driven starting point.

Comment (BenTheElder) [reply release-blocking-jobs.md:159]: /hold
I'm going to come back to ping the thread again when I've incorporate this properly, for now going to get in #2229 first

Comment (aojea) [reply release-blocking-jobs.md:159]: ping @BenTheElder can we put a closure on this?

Comment (BenTheElder) [reply release-blocking-jobs.md:159]: Please fork the PR if you have bandwidth, still too many things going on.

**Pull request closed without merging**